### PR TITLE
HWLP boundary bug, ALU encoding bug and general formatting fixes

### LINKED ISSUE #36: gas: encoding of cv.ext* and hardware loop constants
A while ago I rewrote and ported the pulp gcc and pulp binutils to a newer gcc release. I imported the new tests that were added to the assembler and I found a few discrepancies:

From https://cv32e40p.readthedocs.io/en/latest/instruction_set_extensions/#general-alu-encoding
* `cv.extbs t0,t1` is encoded as `0x040362b3`, but I think this should be `0x100362b3`. The 28-th bit (counting from zero) should be 1. (`python -c 'print(0x100362b3 >> 28 & 1)'`). Similar issue with the other `ext*` instructions.

From https://cv32e40p.readthedocs.io/en/latest/instruction_set_extensions/#hardware-loops
* In the assembler and in the test `cv-hwlp-fail-operand-01.s` it assumes that the branch offset must be an even integer in `[0, 4094]`. Looking at the encoding though (12-bit unsigned immediate) it seems to me the range should be an even integer in `[0, 8190]`

Robert

### PR DESCRIPTION
**Pull request by jessicamills**
- Fix for bug relating to incorrect encoding of ALU sign and zero extend operations.
- Fix for bug relating to incorrect HWLP branch offset upper boundaries.
- Fix general white space and number formatting inconsistencies.

gas/ChangeLog.COREV:

	* config/tc-riscv.c (riscv_ip): Fix boundary bug for hardware loop
	branch operands - upper boundary of long branch offset changed from
	4094 to 8191 and short offset from 30 to 63.

gas/testsuite/ChangeLog.COREV:

	* gas/riscv/cv-alu-extbs.d: Fix test after encoding change.
	* gas/riscv/cv-alu-extbs.s: Likewise.
	* gas/riscv/cv-alu-extbz.d: Likewise.
	* gas/riscv/cv-alu-exths.d: Likewise.
	* gas/riscv/cv-alu-exthz.d: Likewise.
	* gas/riscv/cv-alu-march-rv32i-xcorev.d: Amend test with branch
	offset boundary change.
	* gas/riscv/cv-hwlp-endi.d: Likewise.
	* gas/riscv/cv-hwlp-endi.s: Likewise.
	* gas/riscv/cv-hwlp-fail-operand-01.l: Likewise.
	* gas/riscv/cv-hwlp-fail-operand-01.s: Likewise.
	* gas/riscv/cv-hwlp-fail-operand-02.l: Likewise.
	* gas/riscv/cv-hwlp-fail-operand-02.s: Likewise.
	* gas/riscv/cv-hwlp-fail-operand-03.l: Likewise.
	* gas/riscv/cv-hwlp-fail-operand-03.s: Likewise.
	* gas/riscv/cv-hwlp-fail-operand-04.l: Likewise.
	* gas/riscv/cv-hwlp-fail-operand-04.s: Likewise.
	* gas/riscv/cv-hwlp-setup.d: Likewise.
	* gas/riscv/cv-hwlp-setup.s: Likewise.
	* gas/riscv/cv-hwlp-starti.d: Likewise.
	* gas/riscv/cv-hwlp-starti.s: Likewise.

include/ChangeLog.COREV:

	* opcode/riscv-opc.h: Fix hexadecimal number format inconsistencies
	for matches and masks.

ld/testsuite/ChangeLog.COREV:

	* ld-riscv-elf/ld-riscv-elf.exp: Renamed CORE-V hwlp tests to match
	-march option for the instruction set.
	* ld-riscv-elf/cv-hwloop-starti.s: Renamed to cv-hwlp-starti.s.
	* ld-riscv-elf/cv-hwloop-starti.d: Renamed to cv-hwlp-starti.d.
	* ld-riscv-elf/cv-hwloop-endi.d: Renamed to cv-hwlp-endi.d.
	* ld-riscv-elf/cv-hwloop-endi.s: Renamed to cv-hwlp-endi.s.
	* ld-riscv-elf/cv-hwloop-setup.d: Renamed to cv-hwlp-setup.d.
	* ld-riscv-elf/cv-hwloop-setup.s: Renamed to cv-hwlp-setup.s.
	* ld-riscv-elf/cv-hwloop-setupi.d: Renamed to cv-hwlp-setupi.d.
	* ld-riscv-elf/cv-hwloop-setupi.s: Renamed to cv-hwlp-setupi.s.

opcodes/ChangeLog.COREV:

	* riscv-opc.c (MATCH_CV_EXTHS, MATCH_CV_EXTHZ, MATCH_CV_EXTBS)
	(MATCH_CV_EXTBZ): Add new match.
	(riscv_opcodes): Fix indentation inconsistencies.

### CODE REVIEW

**Comment by jessicamills** [comment]
@NBruschi & @EnricoTabanelli could you please review this fix. Thank you

**Comment by jessicamills** [comment]
@NBruschi & @EnricoTabanelli I have now rebased this after the merge of your work, please review when possible. Thanks

File: ld/testsuite/ld-riscv-elf/cv-hwlp-endi.d [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #name: endi relocation
-#source: cv-hwloop-endi.s
+#source: cv-hwlp-endi.s
```
Comment (NBruschi) [inline ld/testsuite/ld-riscv-elf/cv-hwlp-endi.d:2]: You should modify also the test file located in ld/testsuite/ld-riscv-elf/ld-riscv-elf.exp

**Comment by NBruschi** [comment]
@jessicamills, All looks good but I noticed an ld regression, from 38 expected passes to 34 and 4 unresolved. They are because of you changed the names of the ld hwloop tests from *hwloop* to *hwlp*. You should also edit the test file located in ld/testsuite/ld-riscv-elf/ld-riscv-elf.exp

**Comment by jessicamills** [comment]
Thank you @NBruschi, I have updated this. Please let me know if all is ok.

**Review by NBruschi — APPROVED** [review]
All looks good

**Comment by jessicamills** [comment]
@NBruschi Are you able to merge?

**Comment by NBruschi** [comment]
Unfortunately I have not the write access to merge this pull request

**Comment by jessicamills** [comment]
Ok no problem - since you have reviewed and approved this pull request, I will merge it.

**Pull request merged**